main.cpp

```diff
@@ -58,46 +58,83 @@ Purpose:  This project will show you the difference between member functions and
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    int value = 0; // 2
+    std::string name = ""; // 3
+
+    T (int val, const char* charPtr)   // 1
+    {
+        value = val;
+        name = charPtr;
+    }
```

Comment:
use the constructor member initializer list, not in-constructor initialization.
The reason: the values are being initialized when the class is created.  Then, they're being assigned to.  That's extra work.   
If you use the constructor member initializer list, they'll only be initialized once.
- [ ] revise